source/agora/common/Config.d

```diff
@@ -312,68 +308,130 @@ private BanManager.Config parseBanManagerConfig (Node node)
     return conf;
 }
 
-/// Parse the quorum config section
-private QuorumConfig[] parseQuorumSection (Node quorums_root)
+/*******************************************************************************
+
+    Parse the quorum config section
+
+    Params:
+        node = the Yaml node position to one quorum configuration
+        level = the nesting level of the quorum. The maximum nesting is 3.
+
+    Returns:
+        the parsed quorum config section
+
+*******************************************************************************/
+
+private QuorumConfig parseQuorumSection (Node node, size_t level = 1)
 {
-    struct PreQuorum
-    {
-        string threshold;
-        PublicKey[] nodes;
-        string[] sub_quorums;  // they're looked up later
-    }
+    import std.algorithm;
+    import std.exception;
+    enforce(level <= 3, "Cannot have more than 2 levels of sub-quorums.");
```

Comment:
Why?

Reply:
I was basing it on Stellar. Maybe I should add it to the notes.

Actually funny thing, they increased the depth recently to enable 3 sub-quorums: https://github.com/stellar/stellar-core/pull/2233/commits/aa45d9eaa4ece2d2d6f9b2bcbdc5898ba5d18080